homeassistant/helpers/entity.py

```diff
@@ -278,8 +295,14 @@ def async_update_ha_state(self, force_refresh=False, context=None):
             # Could not convert state to float
             pass
 
+        if (self._context is not None and
+                dt_util.utcnow() - self._context_set >
+                self.context_recent_time):
+            self._context = None
+            self._context_set = None
```

Comment:
This feel a bit wired but there is actual no better solution

Reply:
We need it to track when the state was set. We can't use when context was created because the context might have been part of a script that contained a delay.